pints/residuals_diagnostics.py

```diff
@@ -11,6 +11,311 @@
 import scipy.special
 
 
+def plot_residuals_binned_autocorrelation(parameters,
+                                          problem,
+                                          thinning=None,
+                                          n_bins=25):
+    r"""
+    Plot the autocorrelation of the residuals within bins.
```

Comment:
Can we change this to "Plot the autocorrelation of the residuals within bins (i.e. discrete time windows across the series)".